axiom/nr_php_packages.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef nr_php_packages_HDR
+#define nr_php_packages_HDR
+
+#include "nr_log_event.h"
+#include "util_random.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+
+typedef struct _nr_php_package_t {
+  char* package_name;
+  char* package_version;
+} nr_php_package_t;
+
+typedef nr_hashmap_t nr_php_packages_t;
+
+/*
+ * Purpose : Create a new php package
+ *
+ * Params  : 1. Package name
+ *           2. Package version
+ * 
+ * Returns : A php package that has a name and version. If nr_php_packages_add_package()
+ *           is not called, then it must be freed by nr_php_package_destroy()
+ */
+extern nr_php_package_t* nr_php_package_create(char* name, char* version);
+
+/*
+ * Purpose : Destroy/free php package
+ *
+ * Params  : The php package to free
+ * 
+ * Returns : Nothing, it is void
+ */
+extern void nr_php_package_destroy(nr_php_package_t* p);
+
+/*
+ * Purpose : Add new php package to hashmap. If the hashmap does not exist and this is the first package
+ *           being added, then a new hashmap will be created by calling nr_hashmap_create(). This function
+ *           also frees the package being added, so the user does not need to call nr_php_packages_destroy_package()
+ *
+ * Params  : 1. A pointer to the pointer of a hashmap where packages will be added
+ *           2. A pointer to the php package that needs to be added to hashmap
+ * 
+ * Returns : Nothing, it is void
+ */
+extern void nr_php_packages_add_package(nr_php_packages_t** h, nr_php_package_t* p);
+
+/*
+ * Purpose : Destroy/free the hashmap
+ *
+ * Params  : 1. A pointer to the pointer of a hashmap
+ * 
+ * Returns : Nothing, it is void
+ */
+extern void nr_php_packages_destroy_package(nr_php_packages_t** h);
```

Comment:
It destroys the packages hashmap, not a single package, right? I so then I suggest function name change:
```suggestion
extern void nr_php_packages_destroy(nr_php_packages_t** h);
```

Reply:
Yeah good idea, fixed: [ca1b1d7](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/ca1b1d7f5cfb31c2a07c983497c1871fefccc613)